### ISSUE TITLE
Add a ShuffleSplit method

### ISSUE THREAD

**Issue by vchabri** [openturns/openturns#2583]
### What is the idea?

It would be valuable and useful to have a "ShuffleSplit" method such as the one available in Scikit-learn in order to split a Sample into, e.g., two subsamples based on a random shuffling strategy.

### Why is this needed?

Such a task is very useful for metamodel training and validation.

### Additional Context

Here are the main references:

- [Doc page](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html#sklearn.model_selection.ShuffleSplit.split)
- [Source code](https://github.com/scikit-learn/scikit-learn/blob/f07e0138b/sklearn/model_selection/_split.py#L1781)

**Comment by mbaudin47** [comment]
A workaround is to use the method presented in: http://openturns.github.io/openturns/latest/auto_data_analysis/manage_data_and_samples/plot_randomize_sample_lines.html
